### ISSUE TITLE
ids getting updated too quickly

### ISSUE THREAD

**Issue by maylad31** [Qidian213/deep_sort_yolov3#55]
hi,first of all nice code.Thanks. However in some videos it updates the ids very quickly not even seconds pass and i get id 23,38 etc. Can you tell me why?

**Comment by Qidian213** [comment]
First ，there has a bug i forgot to update on github  . it should be `if not track.is_confirmed() or track.time_since_update > 1:` in demo.py file . Second  for the performance ,there  have many place need to improve .eg, it's features extract ability not very good ,may cause unstable ID . I may do something to improve it

**Comment by maylad31** [comment]
Hi,thanks for the quick response. I would like to ask you something. Do you think increasing the size of feature vector in deepsort from 128 to something greater would help. Secondly i am tracking people indoor while the mars dataset has mostly outdoor images. Do you think training on a different dataset would improve performance.

**Comment by Qidian213** [comment]
I think a better loss function is more effective than increase the size of feature vector .For different people have  same apperance , same person have varying apperance . In this case loss function need to penalize more the same apperance to ensure max distance of same person less than min distance of different person . A way give same apperance distance a bigger weight may help improve it .

**Comment by Qidian213** [comment]
For using indoor ,a dataset bese on indoor case may better .

**Comment by maylad31** [comment]
Thanks for your help.You may close this issue.

**Comment by maylad31** [comment]
hi sorry to disturb you again
I have trained my own dataset to generate .pb file using this code
https://github.com/nwojke/cosine_metric_learning
Now can I use this .pb file directly replacing the one given by you?

**Comment by Qidian213** [comment]
why not have a try ?

**Comment by maylad31** [comment]
Hi
Can you please tell me how to do batch inference. I change the batch size in yolo.cfg file but I get an error could not convert batchdescriptor. Do i need to modify the code in yolo.py?

**Comment by Akhtar303nu** [comment]
Hi @maylad31 
I want to train model on vehicles category (car,truck,bus,motorcycle)
How can I do that 
Thanks

**Comment by Akhtar303nu** [comment]
@Qidian213  @maylad31  any update of parameters to avoid Ids getting updated too quickly
Thanks

**Comment by ss199302** [comment]
@maylad31 How effective is the weight of your own training?

**Comment by harshsp31** [comment]
@Qidian213 @maylad31 Did you guys find a solution for track ids getting updated too quickly?

**Comment by ruksin** [comment]
@Qidian213 @maylad31 is there any solution for tracking ids getting updated quickly.?

**Comment by maylad31** [comment]
Hi, let me tel you what you can do to improve the performance of deepsort. You need to set the values of parameters like max age and uncertainties properly depending on your use case. Also replace deepsort reid with a better one.

**Comment by Ashwin-Ramesh2607** [comment]
@maylad31 Hey,It would be better if you could explain what those parameters mean. Also I did not understand your last statement. What do you mean deepsort reid?
